src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,543 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using static System.Runtime.InteropServices.ComWrappers;
+
+namespace System.Runtime.InteropServices
+{
+    // Defined in windows.ui.xaml.hosting.referencetracker.h.
+    internal enum XAML_REFERENCETRACKER_DISCONNECT
+    {
+        // Indicates the disconnect is during a suspend and a GC can be trigger.
+        XAML_REFERENCETRACKER_DISCONNECT_SUSPEND = 0x00000001
+    };
+
+    internal struct HostServices
+    {
+        internal static IntPtr s_globalHostServices = CreateHostServices();
+
+        private static unsafe IntPtr CreateHostServices()
+        {
+            IntPtr wrapperMem = (IntPtr)NativeMemory.Alloc(2 * (nuint)sizeof(IntPtr));
+            *(IntPtr*)(wrapperMem) = DefaultIReferenceTrackerHostVftblPtr;
+            *(IntPtr*)(wrapperMem + sizeof(IntPtr)) = wrapperMem;
+            return wrapperMem;
+        }
+
+        public static void DisconnectUnusedReferenceSources(uint flags)
+        {
+            if ((((XAML_REFERENCETRACKER_DISCONNECT)flags) & XAML_REFERENCETRACKER_DISCONNECT.XAML_REFERENCETRACKER_DISCONNECT_SUSPEND) != 0)
+            {
+                // In CoreCLR, low_memory_p is also set to true which it isn't here in AOT currently via this code path.
```

Comment:
It would be fine to add a new argument to the existing RhCollect export so that the flag can be passed in.